pyomo/contrib/doe/fim_doe.py

```diff
@@ -1775,7 +1906,30 @@ def extract_FIM(self, m, dv_set, result, y_set=None, t_all_set=None, obj=None, a
                     for t, tim in enumerate(t_all_set):
                         JAC[n1, n2, t] = value(m.jac[name1, name2, tim])
         print('JAC is:', JAC)
+
+        jaco_info = {}
```

Comment:
You see to be duplicating code. This makes it harder to maintain code in the long term. Instead, we should generalize this and move it to a function.

Reply:
Let's talk about how to generalize the remove the duplicated code.